### PR TITLE
AB#91004 added infrastructure types to front door project

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: src/HE.Investments.FrontDoor.Domain/Project/CommandHandlers/ProvideInfrastructureTypesCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using HE.Investments.Account.Shared;
+using HE.Investments.Common.Contract.Validators;
+using HE.Investments.FrontDoor.Contract.Project.Commands;
+using HE.Investments.FrontDoor.Domain.Project.Repository;
+using HE.Investments.FrontDoor.Domain.Project.ValueObjects;
+using MediatR;
+
+namespace HE.Investments.FrontDoor.Domain.Project.CommandHandlers;
+
+public class ProvideInfrastructureTypesCommandHandler : IRequestHandler<ProvideInfrastructureTypesCommand, OperationResult>
```
Comment (jjankowski87) [inline src/HE.Investments.FrontDoor.Domain/Project/CommandHandlers/ProvideInfrastructureTypesCommandHandler.cs:10]: Use SiteBaseCommandHandler

File: src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+using HE.Investments.Common.Messages;
+using HE.Investments.FrontDoor.Contract.Project.Enums;
+
+namespace HE.Investments.FrontDoor.Domain.Project.ValueObjects;
+
+public class ProjectInfrastructureTypes : ValueObject, IQuestion
+{
+    public ProjectInfrastructureTypes(IList<InfrastructureType> infrastructureTypes)
+    {
+        if (infrastructureTypes.Any(x => x == InfrastructureType.IDoNotKnow) && infrastructureTypes.Count > 1)
```
Comment (jjankowski87) [inline src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs:13]: This question is required and you're missing such validation

File: src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            OperationResult.New()
+                .AddValidationError("InfrastructureTypes", ValidationErrorMessage.InvalidValue)
```
Comment (jjankowski87) [inline src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs:16]: Please use `ExclusiveOptionSelected` validation error message for consistency.

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investments.FrontDoor.WWW/Views/Project/Infrastructure.cshtml [hunk 4]
```diff
@@ -11,7 +13,13 @@
     @await Html.PartialAsync("~/Partials/Errors/_ErrorSummaryPartial.cshtml")
```
Comment (Antek012) [inline src/HE.Investments.FrontDoor.WWW/Views/Project/Infrastructure.cshtml:13]: missing page caption

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investments.FrontDoor.Domain/Project/ProjectEntity.cs [hunk 5]
```diff
@@ -74,6 +78,11 @@ public void ProvideAffordableHomesAmount(ProjectAffordableHomesAmount affordable
         AffordableHomesAmount = _modificationTracker.Change(AffordableHomesAmount, affordableHomesAmount);
     }
 
+    public void ProvideInfrastructureTypes(ProjectInfrastructureTypes infrastructureTypes)
+    {
+        InfrastructureTypes = _modificationTracker.Change(InfrastructureTypes, infrastructureTypes);
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investments.FrontDoor.Domain/Project/ProjectEntity.cs:95]: Done

File: src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs (hunk 2 above)
Comment (rafalgajewski) [reply src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs:20]: Done

File: src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs (hunk 3 above)
Comment (rafalgajewski) [reply src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs]: Done

File: src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    public static ProjectInfrastructureTypes Empty() => new();
+
+    public bool IsAnswered()
+    {
+        return Values.Any();
+    }
+
+    protected override IEnumerable<object?> GetAtomicValues()
+    {
+        yield return Values;
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investments.FrontDoor.Domain/Project/ValueObjects/ProjectInfrastructureTypes.cs]: Fixed

File: src/HE.Investments.FrontDoor.WWW/Views/Project/Infrastructure.cshtml (hunk 4 above)
Comment (rafalgajewski) [reply src/HE.Investments.FrontDoor.WWW/Views/Project/Infrastructure.cshtml:13]: According to mocks there is no caption on this page

Comment (Carq) [reply src/HE.Investments.FrontDoor.WWW/Views/Project/Infrastructure.cshtml:13]: image

**Pull request merged**